src/Bau/TaskListWriter.cs

```diff
@@ -0,0 +1,218 @@
+﻿// <copyright file="TaskListWriter.cs" company="Bau contributors">
+//  Copyright (c) Bau contributors. (baubuildch@gmail.com)
+// </copyright>
+
+namespace BauCore
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Text.RegularExpressions;
+
+    public class TaskListWriter

```

Comment:
As you commented below, the implementation may not be ideal yet but so long as it does it's job, it can be refactored later. For now, I'd prefer if this was made `internal`.